presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/Partition.java

```diff
@@ -58,43 +55,40 @@ public Partition(
             @JsonProperty("eligibleToIgnore") boolean eligibleToIgnore,
             @JsonProperty("sealedPartition") boolean sealedPartition,
             @JsonProperty("createTime") int createTime,
-            @JsonProperty("lastDataCommitTime") long lastDataCommitTime)
+            @JsonProperty("lastDataCommitTime") long lastDataCommitTime,
+            @JsonProperty("lastAccessTime") int lastAccessTime)
     {
-        this.databaseName = requireNonNull(databaseName, "databaseName is null");
-        this.tableName = requireNonNull(tableName, "tableName is null");
-        this.values = ImmutableList.copyOf(requireNonNull(values, "values is null"));
+        super(values, databaseName, tableName, createTime, lastAccessTime, makeStorageDescriptor(tableName, columns, storage, new HiveColumnConverter()), parameters);
```

Comment:
Do we need to maintain storage as separate entity in this class or we can derive all required information from sd itself?